src/perftest_resources.c

```diff
@@ -1394,12 +1394,25 @@ int verify_params_with_device_context(struct ibv_context *context,
 		}
 	}
 
-	if (current_dev == CONNECTX3_PRO || current_dev == CONNECTX3)
+	// those are devices supporting new post send
```

Comment:
The correct way to perform fall-back is to do test in configure stage, set some config define after that and use that define in the code.

Definitely not to whiltelist devices which won't allow to use this perftest for future devices.

Thanks

Reply:
I'm not sure this can be resolved in configure time, we can check that rdma-core supports the new API, but how do you check if the provider supports it?

Reply:
In theory, we can write simple application test that will try to use this API and if it gets EOPNOTSUP/ENOSYS, it will mean to us that it is not supported.

If it is not possible to do during configure stage, we can do this test inside perftest too, for first call, and cache the result.

Reply:
I will work on testing inside the perftest, but I also want to state that whitelist is not such a big problem: anyway we have such a whitelist[ here](https://github.com/linux-rdma/perftest/blob/2490003cc8ecc33db053de0fefb023ea191dbec4/src/perftest_parameters.c#L1539).

Reply:
> I will work on testing inside the perftest, but I also want to state that whitelist is not such a big problem: anyway we have such a whitelist[ here](https://github.com/linux-rdma/perftest/blob/2490003cc8ecc33db053de0fefb023ea191dbec4/src/perftest_parameters.c#L1539). And this whitelist is very important too, because we cannot do some stuff for unknown devices (such as inline).

One of the important goals of the library and various capabilities is to avoid whitelists.

Reply:
> > I will work on testing inside the perftest, but I also want to state that whitelist is not such a big problem: anyway we have such a whitelist[ here](https://github.com/linux-rdma/perftest/blob/2490003cc8ecc33db053de0fefb023ea191dbec4/src/perftest_parameters.c#L1539). And this whitelist is very important too, because we cannot do some stuff for unknown devices (such as inline).
> 
> One of the important goals of the library and various capabilities is to avoid whitelists.

I understand. Do we have a cap for ibv_wr_* api support?

Reply:
> > > I will work on testing inside the perftest, but I also want to state that whitelist is not such a big problem: anyway we have such a whitelist[ here](https://github.com/linux-rdma/perftest/blob/2490003cc8ecc33db053de0fefb023ea191dbec4/src/perftest_parameters.c#L1539). And this whitelist is very important too, because we cannot do some stuff for unknown devices (such as inline).
> > 
> > 
> > One of the important goals of the library and various capabilities is to avoid whitelists.
> 
> I understand. Do we have a cap for ibv_wr_* api support?

To some extend - yes.
Try to create QP with IBV_QP_INIT_ATTR_SEND_OPS_FLAGS flag, if you get an error, I think that it should be EOPNOTSUPP, you will know that it is not supported.